sdk/storage/azure-storage-blobs/src/blob_client.cpp

```diff
@@ -191,10 +191,7 @@ namespace Azure { namespace Storage { namespace Blobs {
         {
           newOptions.Range.Value().Length = options.Range.Value().Length.Value() - retryOffset;
         }
-        if (!newOptions.AccessConditions.IfMatch.HasValue())
-        {
-          newOptions.AccessConditions.IfMatch = eTag;
-        }
+        newOptions.AccessConditions.IfMatch = eTag;
```

Comment:
> this PR is to fix a data consistency issue where the blob/file is overwritten during a download process that requires more than one HTTP request. 

Help me understand how such change fixes this issue?
How did you test it?

Reply:
Before this change, customers can specify `*` for this `IfMatch` value. In that case, we do not throw exception even if the ETag from multiple HTTP requests doesn't match.

Reply:
> How did you test it?

It's not easy to test. I feel confident about my code.